tests/tracer/test_sampler.py

```diff
@@ -903,8 +947,13 @@ def test_datadog_sampler_tracer_start_span(dummy_tracer):
 
     spans = dummy_tracer.pop()
     assert len(spans) == 1, "Span should have been sampled and written"
-    assert spans[0].get_metric(SAMPLING_PRIORITY_KEY) is USER_KEEP
-    assert_sampling_decision_tags(spans[0], rule=1.0, limit=None)
+    assert_sampling_decision_tags(
+        spans[0],
+        rule=1.0,
+        limit=None,
+        sampling_priority=USER_KEEP,
```

Comment:
So is this the behavior we'd expect before sending a payload to the agent? If so, why does it not align with here: https://github.com/DataDog/dd-trace-py/pull/3797/files#r898285280
Ah, because we don't set SamplingRule(sample_rate=1.0) for that one?